.buildkite/scripts/steps/upload_to_s3.sh

```diff
@@ -9,7 +9,7 @@
 # limitation.
 #
 # If this isn't a PR build and isn't a debug build then upload the artifacts
-if [[ -z "$BUILDKITE_PULL_REQUEST" && -z "$ML_DEBUG" ]] ; then
+if [[ -z "$BUILDKITE_PULL_REQUEST" || x"$BUILDKITE_PULL_REQUEST" = "xfalse" && -z "$ML_DEBUG" ]] ; then
```

Comment:
This is a very complex condition, mixing `||` and `&&` when the operator precedence of those varies between `[` `]` and `[[` `]]`.

I think it would be better to refactor it to take out that complexity.

```suggestion
if [[ x"$BUILDKITE_PULL_REQUEST" != xtrue && -z "$ML_DEBUG" ]] ; then
```

Plus also add a comment to say that we think BuildKite always sets `BUILDKITE_PULL_REQUEST` but want to treat not set as equivalent to `false`.